libpod/rootless_cni_linux.go

```diff
@@ -258,9 +258,24 @@ func startRootlessCNIInfraContainer(ctx context.Context, r *Runtime) (*Container
 		Options:     []string{"ro"},
 	}
 	g.AddMount(etcCNINetD)
-	// FIXME: how to propagate ProcessArgs and Envs from Dockerfile?
-	g.SetProcessArgs([]string{"sleep", "infinity"})
-	g.AddProcessEnv("CNI_PATH", "/opt/cni/bin")
+
+	inspectData, err := newImage.Inspect(ctx)
+	if err != nil {
+		return nil, err
+	}
+	imageEnv, err := env.ParseSlice(inspectData.Config.Env)
+	if err != nil {
+		return nil, err
+	}
+	for k, v := range imageEnv {
+		g.AddProcessEnv(k, v)
+	}
+	if len(inspectData.Config.Cmd) > 0 {
+		g.SetProcessArgs(inspectData.Config.Cmd)
+	} else {
+		g.SetProcessArgs([]string{"sleep", "infinity"})
```

Comment:
I think we should rather error out when `len(inspectData.Config.Cmd) == 0`

Reply:
Sounds good :+1: